manifests/init.pp

```diff
@@ -64,115 +79,179 @@
+#                       type:String
+#
 class katello (
-  $user             = $katello::params::user,
-  $group            = $katello::params::group,
-  $user_groups      = $katello::params::user_groups,
-
-  $oauth_key        = $katello::params::oauth_key,
-  $oauth_secret     = $katello::params::oauth_secret,
-
-  $post_sync_token  = $katello::params::post_sync_token,
-  $num_pulp_workers = $katello::params::num_pulp_workers,
-  $max_tasks_per_pulp_worker = $katello::params::max_tasks_per_pulp_worker,
-  $log_dir          = $katello::params::log_dir,
-  $config_dir       = $katello::params::config_dir,
-  $proxy_url        = $katello::params::proxy_url,
-  $proxy_port       = $katello::params::proxy_port,
-  $proxy_username   = $katello::params::proxy_username,
-  $proxy_password   = $katello::params::proxy_password,
-  $pulp_max_speed   = $katello::params::pulp_max_speed,
-  $cdn_ssl_version  = $katello::params::cdn_ssl_version,
-
-  $package_names    = $katello::params::package_names,
-  $enable_ostree    = $katello::params::enable_ostree,
-
-  $repo_export_dir  = $katello::params::repo_export_dir,
+  $user                      = $::katello::params::user,
+  $group                     = $::katello::params::group,
+  $user_groups               = $::katello::params::user_groups,
+
+  $oauth_key                 = $::katello::params::oauth_key,
+  $oauth_secret              = $::katello::params::oauth_secret,
+
+  $post_sync_token           = $::katello::params::post_sync_token,
+  $num_pulp_workers          = $::katello::params::num_pulp_workers,
+  $max_tasks_per_pulp_worker = $::katello::params::max_tasks_per_pulp_worker,
+  $log_dir                   = $::katello::params::log_dir,
+  $config_dir                = $::katello::params::config_dir,
+  $proxy_url                 = $::katello::params::proxy_url,
+  $proxy_port                = $::katello::params::proxy_port,
+  $proxy_username            = $::katello::params::proxy_username,
+  $proxy_password            = $::katello::params::proxy_password,
+  $pulp_max_speed            = $::katello::params::pulp_max_speed,
+  $cdn_ssl_version           = $::katello::params::cdn_ssl_version,
+
+  $package_names             = $::katello::params::package_names,
+  $enable_ostree             = $::katello::params::enable_ostree,
+
+  $repo_export_dir           = $::katello::params::repo_export_dir,
+
+  $enable_candlepin          = $::katello::params::enable_candlepin,
+  $enable_qpid               = $::katello::params::enable_qpid,
+  $enable_qpid_client        = $::katello::params::enable_qpid_client,
+  $enable_pulp               = $::katello::params::enable_pulp,
+  $enable_katello            = $::katello::params::enable_katello,
+
+  $candlepin_host            = $::katello::params::candlepin_host,
+  $pulp_host                 = $::katello::params::pulp_host,
+  $qpid_host                 = $::katello::params::qpid_host,
+  $candlepin_url             = "https://${candlepin_host}:8443/candlepin",
```

Comment:
I think it should evaluate from top to bottom, so that should work, but I wanted to try to write a spec test for it :)
I had to take it out of the params though, that should not work when url is in params and host is changed in init.pp -- at least I think so :)